java/source/io.rst

```diff
@@ -0,0 +1,354 @@
+========================
+Reading and writing data
+========================
+
+Recipes related to reading and writing data from disk using
+Apache Arrow.
+
+.. contents::
+
+Writing array
+=============
+
+It is possible to dump data in the raw arrow format which allows 
+direct memory mapping of data from disk. This format is called
+the Arrow IPC format. There are two option: Random access format
+& Streaming format.
```

Comment:
Can we explain or link to docs about the difference between the two?

Also I think we usually call the "random access format" the "file" format (vs the "stream" format), especially since it's called the file format below.

Reply:
Changed